### PR TITLE
Fix `mj_sizeModel`

### PR DESCRIPTION
**Pull request by hartikainen**
Before these changes, the included `EngineIoTest.VerifySizeModel` test fails with the following:
```
mujoco/test/engine/engine_io_test.cc:79: Failure
Expected equality of these values:
  file_size
    Which is: 2416
  mj_sizeModel(model)
    Which is: 6488
```

With the fix, the test passes.

### CODE REVIEW

**Review by saran-t — APPROVED** [review]

**Comment by saran-t** [comment]
This is failing on MSVC..

File: test/engine/engine_io_test.cc [hunk 1]
```diff
@@ -47,6 +51,34 @@ mjModel PartialModel(const mjModel* m) {
...
+      </body>
+    </worldbody>
+  </mujoco>
+  )";
+
+  std::array<char, 1024> error;
+  mjModel* model = LoadModelFromString(xml, error.data(), error.size());
+  ASSERT_THAT(model, NotNull()) << "Failed to load model: " << error.data();
+
+  std::filesystem::path temp_file = (
```
Comment (nimrod-gileadi) [inline test/engine/engine_io_test.cc:70]: This test fails on Windows (with MSVC). We have another test that creates temporary files like this [here](https://github.com/deepmind/mujoco/blob/c17375f/test/fixture.cc#L112).

Maybe you can find out why that utility works and this test doesn't? Maybe you'll need to explicitly convert the `std::filesystem::path` it to a `std::string` by calling `.string()`.

**Review by nimrod-gileadi — CHANGES_REQUESTED** [review]

File: test/engine/engine_io_test.cc (hunk 1 above)
Comment (hartikainen) [reply test/engine/engine_io_test.cc:70]: You were right, adding `.string()` fixed the issue.

**Review by saran-t — APPROVED** [review]

File: test/engine/engine_io_test.cc [hunk 2]
```diff
@@ -47,6 +51,34 @@ mjModel PartialModel(const mjModel* m) {
...
+    std::filesystem::temp_directory_path() / "model.mjb");
+
+  mj_saveModel(model, temp_file.string().c_str(), NULL, 0);
+
+  std::uintmax_t file_size = std::filesystem::file_size(temp_file);
+
+  std::filesystem::remove(temp_file);
+
+  EXPECT_EQ(file_size, mj_sizeModel(model));
+}
```
Comment (saran-t) [inline test/engine/engine_io_test.cc:80]: The test is leaking `mjModel` (picked up by the leak checker internally). Please fix.

**Review by saran-t — CHANGES_REQUESTED** [review]

File: test/engine/engine_io_test.cc (hunk 2 above)
Comment (hartikainen) [reply test/engine/engine_io_test.cc:82]: Done.

**Review by saran-t — APPROVED** [review]

**Pull request merged**